src/storage.js

```diff
@@ -4,6 +4,19 @@ const _ = require('lodash');
  * Append progress to deployments.
  */
 module.exports = function(storage, progress) {
+  // Trimming the history to a maximum of 10 records and 490KB (keeping a 10KB buffer).
+  function trimLogs(data) {
+    const maximumBytes = 490000;
+    var dataSize = Buffer.from(JSON.stringify(data)).length;
+
+    while (dataSize >= maximumBytes || data.deployments.length > 10) {
+      data.deployments = _.drop(data.deployments, 1);
```

Comment:
This would kill all the way to zero if the first log is too large.  I'm assuming this would be a good thing, though it might be confusing for the interface, might want to log an error if that is the case.

Reply:
Yes, this is by design.  I pointed that out in the description.  In this instance they have outgrown the usage of the extension and I am not sure if that is a valid use case or something we should consider at this point.  If they have this much to deploy they should be using the APIs directly or the deploy cli.